### PR TITLE
[DEV-5735] metrics calc jira multiple groups

### PR DESCRIPTION
**Pull request by gaetano-guerriero**
Support in `MetricEntriesCalculator`, no input/output

- - -
This change is [Reviewable](https://reviewable.io/reviews/athenianco/athenian-api/3351)

### CODE REVIEW

File: server/athenian/api/internal/miners/jira/issue.py [hunk 1]
```diff
@@ -417,7 +417,7 @@ async def _fetch_jira_issues(
         if (missing_updated := issues[AthenianIssue.updated.name].isnull().values).any():
             log.error(
                 "JIRA issues are missing in jira.athenian_issue: %s",
-                ", ".join(issues[Issue.key.name].values[missing_updated]),
+                ", ".join(map(str, issues.index.values[missing_updated])),
```
Comment (gaetano-guerriero) [inline server/athenian/api/internal/miners/jira/issue.py:420]: 'key' column is not  not always present

**Comment by vmarkovtsev** [comment]
Overall, LGTM :+1:

**Review by gaetano-guerriero — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/athenianco/athenian-api/3351)* status: 0 of 11 files reviewed, 2 unresolved discussions (waiting on @vmarkovtsev)

___
*[`server/athenian/api/internal/features/entries.py` line 1246 at r1](https://reviewable.io/reviews/athenianco/athenian-api/3351#-NOEBsLJ3z5Q9ggYzg6f:-NOEJnATElJvtXkC_8Oy:by8pasg) ([raw file](https://github.com/athenianco/athenian-api/blob/145559b9caa3de3918fd2bc197efca95b290eb0c/server/athenian/api/internal/features/entries.py#L1246)):*
<details><summary><i>Previously, vmarkovtsev (Vadim Markovtsev) wrote…</i></summary><blockquote>

`jira_ids` are globally unique and may not change. It is always the same for all the groups and should be left here
</blockquote></details>

I could add it back but where will it be used ?
The problem is the dual nature of `JIRAFilter` which carries both the user filtering and the account status, so someway replace `JIRAConfig`
`fetch_jira_issue` already does not receive `JIRAConfig`
To convert entities for the grouping `JIRAFilter`s are used, only `account` is needed.
`calc_pull_request_metrics_line_github` already accepts only `JIRAFilter` and no `JIRAConfig` given that the needed information are carried by `JIRAFilter`

**Review by vmarkovtsev — CHANGES_REQUESTED** [review]
> *[Reviewable](https://reviewable.io/reviews/athenianco/athenian-api/3351)* status: 0 of 11 files reviewed, 3 unresolved discussions (waiting on @gaetano-guerriero)

___
*[`server/athenian/api/internal/features/entries.py` line 1251 at r2](https://reviewable.io/reviews/athenianco/athenian-api/3351#-NOEr15b3ZKfqDs_nqhY:-NOEr15b3ZKfqDs_nqhZ:bj1hgdg) ([raw file](https://github.com/athenianco/athenian-api/blob/a6f4adcdd4574b2f0fbf273f43dd63f9ff63c067/server/athenian/api/internal/features/entries.py#L1251)):*
> ```Python
>         # there should be at least one group, and every should include a valid jira account id
>         assert len(groups)
>         assert all(g.account for g in groups)
> ```

This is not enough: all the accounts must be exactly equal.
___
*[`server/athenian/api/internal/features/entries.py` line 1262 at r2](https://reviewable.io/reviews/athenianco/athenian-api/3351#-NOErLx77Zh4mwCG17B0:-NOErLx77Zh4mwCG17B1:b-d132no) ([raw file](https://github.com/athenianco/athenian-api/blob/a6f4adcdd4574b2f0fbf273f43dd63f9ff63c067/server/athenian/api/internal/features/entries.py#L1262)):*
> ```Python
>             assert len(groups) == 1
>             extra_columns.append(Issue.labels)
>             general_grouper = label_splitter
> ```

Let's name it `jira_filter_grouper`

**Review by gaetano-guerriero — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/athenianco/athenian-api/3351)* status: 0 of 11 files reviewed, 3 unresolved discussions (waiting on @vmarkovtsev)

___
*[`server/athenian/api/internal/features/entries.py` line 1251 at r2](https://reviewable.io/reviews/athenianco/athenian-api/3351#-NOEr15b3ZKfqDs_nqhY:-NOEtYGQE9707Rcja3BR:b-896fix) ([raw file](https://github.com/athenianco/athenian-api/blob/a6f4adcdd4574b2f0fbf273f43dd63f9ff63c067/server/athenian/api/internal/features/entries.py#L1251)):*
<details><summary><i>Previously, vmarkovtsev (Vadim Markovtsev) wrote…</i></summary><blockquote>

This is not enough: all the accounts must be exactly equal.
</blockquote></details>

Done.
___
*[`server/athenian/api/internal/features/entries.py` line 1262 at r2](https://reviewable.io/reviews/athenianco/athenian-api/3351#-NOErLx77Zh4mwCG17B0:-NOEtXkG7OB2QVOu5Wbc:b-896fix) ([raw file](https://github.com/athenianco/athenian-api/blob/a6f4adcdd4574b2f0fbf273f43dd63f9ff63c067/server/athenian/api/internal/features/entries.py#L1262)):*
<details><summary><i>Previously, vmarkovtsev (Vadim Markovtsev) wrote…</i></summary><blockquote>

Let's name it `jira_filter_grouper`
</blockquote></details>

Done.

**Review by vmarkovtsev — APPROVED** [review]
> *[Reviewable](https://reviewable.io/reviews/athenianco/athenian-api/3351)* status: 0 of 11 files reviewed, 1 unresolved discussion

**Review by vmarkovtsev — APPROVED** [review]
> *[Reviewable](https://reviewable.io/reviews/athenianco/athenian-api/3351)* status: 0 of 11 files reviewed, 1 unresolved discussion

**Review by vmarkovtsev — APPROVED** [review]
> *[Reviewable](https://reviewable.io/reviews/athenianco/athenian-api/3351)* status: 0 of 11 files reviewed, all discussions resolved

**Pull request merged**